src/us/kbase/narrativejobservice/sdkjobs/SDKLocalMethodRunner.java

```diff
@@ -88,25 +55,57 @@
     public static final String CFG_PROP_AWE_CLIENT_CALLBACK_NETWORKS =
             JobRunnerConstants.CFG_PROP_AWE_CLIENT_CALLBACK_NETWORKS;
 
+
+    public static long milliSecondsToLive(String token, Map<String, String> config) throws Exception {
+        String authUrl = config.get(NarrativeJobServiceServer.CFG_PROP_AUTH_SERVICE_URL_V2);
+        if (authUrl == null) {
+            throw new IllegalStateException("Deployment configuration parameter is not defined: " +
+                    NarrativeJobServiceServer.CFG_PROP_AUTH_SERVICE_URL_V2);
+        }
+
+        //Check to see if http links are allowed
+        String authAllowInsecure = config.get(NarrativeJobServiceServer.CFG_PROP_AUTH_SERVICE_ALLOW_INSECURE_URL_PARAM);
+        if (!"true".equals(authAllowInsecure) && !authUrl.startsWith("https://")) {
+            throw new Exception("Only https links are allowed: " + authUrl);
+        }
+        CloseableHttpClient httpclient = HttpClients.createDefault();
+        HttpGet request = new HttpGet(authUrl);
+        request.setHeader(HttpHeaders.AUTHORIZATION, token);
+        InputStream response = httpclient.execute(request).getEntity().getContent();
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> jsonMap = mapper.readValue(response, Map.class);
+        Object expire = jsonMap.getOrDefault("expires", null);
+        //Calculate ms till expiration
+        if (expire != null) {
```

Comment:
I think I would switch the logic around here so that it is

if (expire == null) {
   throw...
}
return (....

Reply:
done